### PR TITLE
Adds `AuthLoadingSwitch` to Navigate different directions, depending on if a user is logged in.

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Adds and implements an AuthLoadingSwitchComponent, which navigates different ways if a user is logged in. When this component is the default route of a navigator, it will wait for a query to return describing if a user is logged in (including waiting for the cache to load) and then navigate to a named route described by the component. 

This is used to handle navigating correctly when the app mounts. To the "home" if the user is logged in, and to the "auth screen" if the user isn't logged in.

Here's an example of usage

```
const AuthLoading = (props) => <AuthLoadingSwitch loggedInRouteName="Home" loggedOutRouteName="Auth" LoadingIndicator={LoadingComponent}

const AppNavigator = createStackNavigator(
  {
    AuthLoading,
    Home,
    Auth,
  },
  {
    initialRouteName: 'AuthLoading',
  }
);
```

### What design trade-offs/decisions were made?

Chose to create this component in the `ui-auth` package to reduce the amount of code that needed to be copy/pasted.

### How do I test this PR?

1. Open the app when logged out, refresh the app, you should end up on the auth screen.
2. Open the app when logged in, refresh the app, you should end up on the tab view.

### What automated tests did you write?

Snapshot tests over new components.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] References #519 
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [x] Manual QA on iOS and ensure it looks/behaves as expected
- [x] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [x] Read through the "Files changed" tab _very carefully_
- [x] Edge cases: what assumptions are made about input?
- [x] What kind of tests could be written?
- [x] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-auth/src/index.js [hunk 1]
```diff
@@ -70,6 +71,7 @@ const CancelButton = styled(({ theme }) => ({
 class Auth extends PureComponent {
   static navigationOptions = {
     header: null,
+    gesturesEnabled: false,
```
Comment (burkeshartsis) [inline packages/apollos-ui-auth/src/index.js]: I'm not sure we want to default this for everywhere auth is used in the app. I agree we need it for how we are using it in this PR though. At the same time, we are transitioning from an app that you auth when you need it to an app that forces auth up front. Either way, I feel like this component should be more flexible than restrictive. Also email login here is just temporary until we get SMS auth merged in.

File: packages/apollos-ui-auth/src/login/Form.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ class Form extends PureComponent {
             </View>
           </PaddedView>
         </KeyboardAwareScrollView>
-        <BottomSafeAreaView>
+        <BottomSafeAreaView forceInset={{ bottom: 'always' }}>
```
Comment (burkeshartsis) [inline packages/apollos-ui-auth/src/login/Form.js]: This is broken on Android. I'm sure it's broken in lots of places. We should probably handle this in a seperate PR but I wanted to mention it.

File: packages/apollos-ui-auth/src/index.js (hunk 1 above)
Comment (burkeshartsis) [reply packages/apollos-ui-auth/src/index.js]: This is the only real question to address before merge.

Comment (vinnyjth) [reply packages/apollos-ui-auth/src/index.js]: @burkeshartsis Not worried about this at this point, we can still hit cancel to close. It's a good callout to make however. 

If `react-navigation` made configuring this option easier, I would be all for it, but as of right now we'll have to work some magic in order for this setting to be configurable via a prop.

**Pull request merged**